### DISCUSSION TITLE
Window Masks

### DISCUSSION THREAD

**Discussion by Cardellino** [linuxdeepin/developer-center#3477] [Q&A | 问答板块] [answered]
Hello, after today's update, the shifting of the window masks no longer works.
My system is:
ArchLinux x86_64
Kernel: 6.0.2-arch1-1
Deepin 20.7.1.

Many thanks in advance
![Bildschirmaufnahme_Auswahlbereich_20221028094225](https://user-images.githubusercontent.com/73848158/198532163-2821a00c-bcd6-428e-bc8c-333f7d567a6f.png)

**Comment by BLumia** [comment]
I don't understand what does "Window Masks" means, could you be more specific so we can better understand what went wrong? Thanks!

**Reply by Cardellino** [reply]
I can't move the window and the buttons are also missing: close, reduce window and enlarge window....

Thank you Blumia for the support

**Reply by BLumia** [reply]
Oh I see. Does all other non-DDE (non-[CSD](https://en.wikipedia.org/wiki/Client-side_decoration)) application window also have this issue? Did you try logout and login again? If already tried, could you provide the output of `ps -e | grep kwin` from terminal? Thanks!

**Reply by Cardellino** [reply]
Yes, even with non-DDE windows.
For ps -e | grep kwin comes nothing!
Installs kwin-5.26.2.1-1

**Reply by Cardellino** [reply]
For example. a non-DDE program.
Cannot move windows, close etc.
![Bildschirmaufnahme_Auswahlbereich_20221028150024](https://user-images.githubusercontent.com/73848158/198597449-ca3c0c9b-c274-4087-874b-e664afea3e38.png)

**Reply by BLumia** [reply]
Seems the window manager is broken. Are you using deepin-kwin? if you do, you can consider install `deepin-wm` and uninstall `deepin-kwin` for a workaround.

**Reply by Cardellino** [reply]
Good morning Blumia,
no change when uninstalling kwin or installing wm.
The problem persists.

**Reply by BLumia** [reply]
What if you run `deepin-wm --replace &` from a terminal?

**Reply by Cardellino** [reply]
Hello Blumia,
with deepin-wm --replace & then everything is fine.
This is the error message from the console:

~]$
(deepin-wm:3762): GLib-GObject-CRITICAL **: 17:54:43.955: g_value_set_boxed: assertion 'G_VALUE_HOLDS_BOXED (value)' failed

(deepin-wm:3762): GLib-GObject-CRITICAL **: 17:54:43.975: g_value_set_boxed: assertion 'G_VALUE_HOLDS_BOXED (value)' failed
[zeferino@sicily ~]$
(deepin-wm:3762): Gtk-WARNING **: 17:54:44.055: Theme parsing error: gtk-dark.css:243:56: Using one color stop with linear-gradient() is deprecated.

(deepin-wm:3762): Gtk-WARNING **: 17:54:44.055: Theme parsing error: gtk-dark.css:250:56: Using one color stop with linear-gradient() is deprecated.

(deepin-wm:3762): Gtk-WARNING **: 17:54:44.055: Theme parsing error: gtk-dark.css:261:56: Using one color stop with linear-gradient() is deprecated.

(deepin-wm:3762): Gtk-WARNING **: 17:54:44.055: Theme parsing error: gtk-dark.css:268:56: Using one color stop with linear-gradient() is deprecated.
![Bildschirmaufnahme_Auswahlbereich_20221029180247](https://user-images.githubusercontent.com/73848158/198841430-a8cccbdd-a494-4de1-baff-721ff3ea8194.png)

As soon as the console is closed, the same problem as before
Thanks for the support

**Reply by BLumia** [reply]
Oh, you can also use `nohup deepin-wm --replace` as a workaround.

ps. I managed to reproduce this issue. If you haven't clean your package cache, you can also downgrade `startdde` to `startdde-5.9.56-1` as a workaround: `sudo pacman -U /var/cache/pacman/pkg/startdde-5.9.56-1-x86_64.pkg.tar.zst`

**Reply by Cardellino** [reply]
With a downgrade everything works again.
Thank you for your support...

**Reply by BLumia** [reply]
The package maintainer already released a fix (`startdde 5.9.59-2`) so you can safely upgrade the package once the mirror you use synced the packages :)

**Comment by BLumia** [comment, marked as answer]
Dive into the source code and seems it's a bug in `startdde 5.9.59` ([caused by this line](https://github.com/linuxdeepin/startdde/blob/47960a334ef523c004014f208539b19d6522bb42/main.go#L90)). You can use any one of the following as a workaround:

1. Start `deepin-wm` manually via `nohup deepin-wm --replace` from the terminal. *or*
2. Open `dconf-editor`, locate to `/com/deepin/dde/startdde/wm-cmd` and change the value to `/usr/bin/deepin-wm`. *or*
3. Rollback to `startdde-5.9.56-1` if you haven't clear package cache: `sudo pacman -U /var/cache/pacman/pkg/startdde-5.9.56-1-x86_64.pkg.tar.zst`

The bug has been reported to the Arch Linux package maintainer and this issue will hopefully get fixed in the next day with `startdde-5.9.59-2` or so. Once the new package has been released then you'll be able to just upgrade the package and everything will work again! (if you did the 2nd workaround solution, don't forget to restore the change once the package get updated :) )
